### ISSUE TITLE
It's not possible to use Pydantic BaseSettings with a package called with Typer CLI

### ISSUE THREAD

**Issue by tiangolo** [tiangolo/typer-cli#10]
Copied original issue from: https://github.com/tiangolo/typer/issues/91

---

### Describe the bug

I am using `typer` as a package where I get the following error

```python
  .
  .
  File "/root/typer_app/main.py", line 9, in publish
    class Settings(BaseSettings):
  File "pydantic/main.py", line 184, in pydantic.main.ModelMetaclass.__new__
  File "pydantic/typing.py", line 136, in pydantic.typing.resolve_annotations
    if arg is None:
KeyError: 'typer_app.main'
```

### To Reproduce

I am using `typer` as a package

```
/root
| typer_app
  | __init__.py
  | main.py
```
where 

 `main.py`
```python
import typer
from pydantic import BaseSettings

app = typer.Typer()


@app.command()
def publish():
    class Settings(BaseSettings):
        HTTP_ENDPOINT: str

    settings = Settings()
    typer.echo(settings.HTTP_ENDPOINT)

```

* Call it with:

```bash
PYTHONPATH=/root typer typer_app.main run
```

* It outputs:

```python
  .
  .
  File "/root/typer_app/main.py", line 9, in publish
    class Settings(BaseSettings):
  File "pydantic/main.py", line 184, in pydantic.main.ModelMetaclass.__new__
  File "pydantic/typing.py", line 136, in pydantic.typing.resolve_annotations
    if arg is None:
KeyError: 'typer_app.main'
```

* But I expected it to output a validation error since HTTP_ENDPOINT is not set.

### Environment

```bash
Linux Mint - Python 3.8.2
typer==0.1.1
typer-cli==0.0.8
```

**Comment by rkbeatss** [comment]
I could possibly take a look at this

**Comment by kbanc** [comment]
Can I look into this?

**Comment by tiangolo** [comment]
I don't remember if I replied back to this on Discord :grimacing: 

Anyway, if you're still up to the challenge, yes, please! :rocket:

**Comment by kbanc** [comment]
Sounds good @tiangolo! I'll be looking into it this week and report any findings :)

**Comment by dazza-codes** [comment]
As a work-around, it works OK when all the pydantic models are extracted to a separate module from the typer CLI module.  In the example for this issue, it might work with:

```python
import typer
from .models import HTTPSettings

app = typer.Typer()

@app.command()
def publish(http_endpoint: str):
    http_settings = HTTPSettings(endpoint=http_endpoint)
    typer.echo(http_settings.endpoint)
```

```python
# models.py
from pydantic import BaseSettings

class HTTPSettings(BaseSettings):
    endpoint: str = Field(required=True)
```

**Comment by cdcadman** [comment]
I have reproduced this issue on pydantic 1.7.4, typer 0.1.1, and typer-cli 0.0.8, and I resolved it by upgrading pydantic to 1.8.  So I believe this is not a bug in typer-cli.

**Comment by tiangolo** [comment]
Indeed @cdcadman! Thanks for the very complete review! :mag: 

I just checked with a recent version of Pydantic and it seems to work correctly now. I didn't expect that! Haha :sweat_smile:
